src/mpid/ch4/src/ch4i_workq_progress.c

```diff
@@ -12,97 +12,110 @@ extern MPID_Thread_mutex_t MPIDI_workq_lock;
 static int MPIDI_workq_dispatch(MPIDI_workq_elemt_t * workq_elemt)
 {
     int mpi_errno = MPI_SUCCESS;
-    MPIR_Request *req;
     MPI_Datatype datatype, origin_datatype, target_datatype;
 
     MPIR_Assert(workq_elemt != NULL);
 
     switch (workq_elemt->op) {
         case SEND:{
                 struct MPIDI_workq_send *wd = &workq_elemt->params.pt2pt.send;
-                req = wd->request;
```

Comment:
So I think my main concern is that this doubles the request allocation overhead of the workq path. That's something I know folks were keenly aware of when developing workq.

Reply:
I see. Let me think about how to cleanly (cleaner than current) handle pre-allocated requests.